drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/widgets/ScenarioGrid.java

```diff
@@ -216,4 +223,47 @@ protected void appendRows(Simulation simulation) {
     protected void appendRow(int rowIndex, Scenario scenario) {
         ((ScenarioGridModel) model).insertRowGridOnly(rowIndex, new ScenarioGridRow(), scenario);
     }
+
+    @Override
+    public boolean adjustSelection(final SelectionExtension direction, final boolean isShiftKeyDown) {
+        final boolean selectionChanged = super.adjustSelection(direction, isShiftKeyDown);
+
+        scenarioGridPanel.signalRightPanelNoHeaderCellSelected();
+
+        if (getModel().getSelectedHeaderCells().size() > 0) {
+            final GridData.SelectedCell cell = getModel().getSelectedHeaderCells().get(0);
+
+            final int uiColumnIndex = ColumnIndexUtilities.findUiColumnIndex(getModel().getColumns(),
+                                                                             cell.getColumnIndex());
+            final int uiRowIndex = cell.getRowIndex();
+
+            setSelectedColumnAndHeader(uiRowIndex, uiColumnIndex);
+
+            final GridColumn column = getModel().getColumns().get(uiColumnIndex);
+            if (uiRowIndex > 0 && column instanceof ScenarioGridColumn) {
+                final EnableRightPanelEvent enableRightPanelEvent = enableRightPanelEvent((ScenarioGridColumn) column,
+                                                                                          cell,
+                                                                                          uiColumnIndex);
+
+                scenarioGridPanel.signalRightPanelHeaderCellSelected(enableRightPanelEvent);
+            }
+        }
+
+        return selectionChanged;
+    }
+
+    EnableRightPanelEvent enableRightPanelEvent(final ScenarioGridColumn scenarioGridColumn,
```

Comment:
Again can't say more that agree with you, I don't prefer this visibility.

I used it because was not able to provide some unit test for that piece of code. Not saying it is not possible to test otherwise but my knowledge was not enough. 

We use this also in kie-wb-common when testing ui code. I understand it is not ideal, will try to test keeping private once more time, but can promise success.

Reply:
Actually, I think you should replace "package" with "protected", if you want to use property/method in test.
I always use "protected" in all my tests, and it works.
Until now, I never needed (nor find any justification) to use the "default" access.